app/uk/gov/hmrc/transitmovements/controllers/MovementsController.scala

```diff
@@ -216,37 +216,59 @@ class MovementsController @Inject() (
           )
     }
 
-  def updateMovementLargeMessage(movementId: MovementId, triggerId: Option[MessageId] = None): Action[AnyContent] = Action.async(parse.anyContent) {
+  private def attachLargeMessage(movementId: MovementId, triggerId: Option[MessageId] = None): Action[AnyContent] = Action.async(parse.anyContent) {
     implicit request =>
-      (for {
-        messageType                 <- extract(request.headers).asPresentation
-        objectStoreURI              <- extractObjectStoreURI(request.headers)
-        objectStoreResourceLocation <- extractResourceLocation(objectStoreURI)
-        sourceFile                  <- objectStoreService.getObjectStoreFile(objectStoreResourceLocation).asPresentation
-        messageData                 <- messagesXmlParsingService.extractMessageData(sourceFile, messageType).asPresentation
-        received = OffsetDateTime.ofInstant(clock.instant, ZoneOffset.UTC)
-        message = messageFactory
-          .createLargeMessage(
-            messageType,
-            messageData.generationDate,
-            received,
-            triggerId,
-            objectStoreURI
-          )
-        _ <- repo.updateMessages(movementId, message, messageData.mrn, received).asPresentation
-      } yield message.id).fold[Result](
-        baseError => Status(baseError.code.statusCode)(Json.toJson(baseError)),
-        id => Ok(Json.toJson(UpdateMovementResponse(id)))
-      )
+      request.headers.get(Constants.ObjectStoreURI).map(ObjectStoreURI.apply) match {
+        case Some(objectStoreURI) => attachLargeMessageInObjectStore(movementId, triggerId, objectStoreURI)
+        case None                 => attachEmptyMessage(movementId)
+      }
+  }
+
+  private def attachEmptyMessage(movementId: MovementId) = {
+    val received = OffsetDateTime.ofInstant(clock.instant, ZoneOffset.UTC)
+
+    val message = messageFactory.createEmptyMessage(
```

Comment:
I think we should status as Pending while creating empty message

Reply:
It does get set to Pending in the messageFactory.createEmptyMessage method doesn't it?

Reply:
yes